### ISSUE TITLE
Error in loading data set containing one molecule.

### ISSUE THREAD

**Issue by zealseeker** [deepchem/deepchem#912]
Reproduce:
Prepare two files, `test.csv` and `bug.py`. Run the bug.py which loads the dataset that contains only one molecule.
```
#test.csv
Smiles,endpoint
c1ccccc1,1
```
```
#bug.py
import deepchem as dc
featurizer_func = dc.feat.ConvMolFeaturizer()
loader = dc.data.CSVLoader(tasks=['endpoint'],featurizer=featurizer_func,smiles_field='Smiles')
dataset = loader.featurize('test.csv',shard_size=8192)
```
output:
```
Loading raw samples now.
shard_size: 8192
About to start loading CSV from test.csv
Loading shard 1 of size 8192.
Featurizing sample 0
Traceback (most recent call last):
  File "bug.py", line 4, in <module>
    dataset = loader.featurize('test.csv',shard_size=8192)
  File "/home/lmmd_wz/anaconda3/envs/deepchem/lib/python3.5/site-packages/deepchem/data/data_loader.py", line 222, in featurize
    shard_generator(), data_dir, self.tasks, verbose=self.verbose)
  File "/home/lmmd_wz/anaconda3/envs/deepchem/lib/python3.5/site-packages/deepchem/data/datasets.py", line 461, in create_dataset
    for shard_num, (X, y, w, ids) in enumerate(shard_generator):
  File "/home/lmmd_wz/anaconda3/envs/deepchem/lib/python3.5/site-packages/deepchem/data/data_loader.py", line 209, in shard_generator
    assert len(X) == len(ids) == len(y) == len(w)
TypeError: len() of unsized object
```

Generally, X should be a list of ConvMol objects. But when there is only one molecule, it will not be wrapped in a list. Then X is an object which cannot `len`.

**Comment by rbharath** [comment]
Bug confirmed on my machine. We should fix this edge case.

**Comment by peastman** [comment]
I think we can close this.  It was fixed by #916.
